fendermint/actors/gas_market/src/lib.rs

```diff
@@ -8,103 +8,120 @@ use fil_actors_runtime::{actor_dispatch, ActorError};
 use fvm_ipld_encoding::tuple::*;
 use fvm_shared::econ::TokenAmount;
 use fvm_shared::METHOD_CONSTRUCTOR;
+use lazy_static::lazy_static;
 use num_derive::FromPrimitive;
-use num_traits::Zero;
-use std::ops::Mul;
+use std::cmp::Ordering;
 
 #[cfg(feature = "fil-actor")]
 fil_actors_runtime::wasm_trampoline!(EIP1559GasMarketActor);
 
 /// Base fee max change denominator as defined in [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
-const EIP1559_BASE_FEE_MAX_CHANGE_DENOMINATOR: u64 = 8;
+const BASE_FEE_MAX_CHANGE_DENOMINATOR: u64 = 8;
 /// Elasticity multiplier as defined in [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
-const EIP1559_ELASTICITY_MULTIPLIER: u64 = 2;
-/// Initial base fee as defined in [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
-pub const EIP1559_INITIAL_BASE_FEE: u64 = 1_000_000_000;
+const ELASTICITY_MULTIPLIER: u64 = 2;
+lazy_static! {
+    /// Initial base fee as defined in [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
+    static ref INITIAL_BASE_FEE: TokenAmount = TokenAmount::from_atto(1_000_000_000);
+    static ref MINIMAL_BASE_FEE: TokenAmount = TokenAmount::from_atto(100);
+}
```

Comment:
How about we put these constants into an object in state, making them configurable via `SetConstants`?

The downside is that they'll be rewritten on every utilization update, and we'll need to read state on every update, thus leading to some minor overhead, but it's pretty negligible and leads to much better customization, so I think it's a tradeoff worth taking.

You can also take this object in the constructor, and feed these during genesis.

Reply:
Can, the miminal base fee perhaps is constant worthy. They are just BigInt, should be ok. If the constants grow larger, I think asking the state to hold a hamt mapping using CID as key can work too.